### PR TITLE
Corrections, fixes, etc.

### PR DESCRIPTION
**Pull request by bouwew**
All changes:
- Move compressor_state into binary_sensors
- Adam: add missing zigbee_mac to wireless thermostats
- Stretch & Adam: don't show devices without a zigbee_mac, should be orphaned devices
- Harmonize appliance dicts for legacy devices
- Typing improvements
- Fix related test asserts

### CODE REVIEW

File: plugwise/helper.py [hunk 1]
```diff
@@ -469,7 +469,9 @@ def _energy_device_info_finder(self, appliance: etree, appl: Munch) -> Munch:
                 appl.model = version_to_model(hw_version)
             appl.fw = module_data["firmware_version"]
             appl.zigbee_mac = module_data["zigbee_mac_address"]
-            return appl
+            # Filter appliance without zigbee_mac, it's an orphaned device
+            if appl.zigbee_mac:
```
Comment (CoMPaTech) [inline plugwise/helper.py:473]: Now we don't return (i.e. continue with the other if statements) ... shouldn't we `if not: return` and leave the generic return? (i.e. we probably hit L561 or something now)

Comment (bouwew) [reply plugwise/helper.py]: Good point. We should always `return appl`, that's what the calling function expects. Implemented.

Comment (bouwew) [reply plugwise/helper.py]: And, continuing with the other if statement is not needed. They are mutually exclusive.

**Review by CoMPaTech — APPROVED** [review]
🌴 👍

**Pull request merged**